libs/common/src/lib/services/bootstrap.service.spec.ts

```diff
@@ -0,0 +1,84 @@
+import { TestBed } from '@angular/core/testing'
+import {
+  SETTINGS_FIXTURES,
+  SITE_FIXTURES,
+  SiteApiService,
+  UI_FIXTURES,
+  UiApiService,
+} from '@lib/gn-api'
+import { of } from 'rxjs'
+
+import { BootstrapService } from './bootstrap.service'
+
+const siteApiServiceMock = {
+  getSiteOrPortalDescription: jest.fn(() => of(SITE_FIXTURES)),
+  getSettingsSet: jest.fn(() => of(SETTINGS_FIXTURES)),
+}
+
+const uiApiServiceMock = {
+  getUiConfiguration: jest.fn((uiIdentifier) => of(UI_FIXTURES)),
+}
+
+describe('BootstrapService', () => {
+  let service: BootstrapService
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        {
+          provide: SiteApiService,
+          useValue: siteApiServiceMock,
+        },
+        {
+          provide: UiApiService,
+          useValue: uiApiServiceMock,
+        },
+      ],
+    })
+    service = TestBed.inject(BootstrapService)
+  })
+
+  it('should be created', () => {
+    expect(service).toBeTruthy()
+  })
+
+  describe('uiConfReady', () => {
+    let uiConf
+    describe('When the service is initialized', () => {
+      it('no conf is stored ', () => {
+        expect(Object.keys(service['uiConfigurations']).length).toBe(0)
+      })
+      it('api has never been called ', () => {
+        expect(service['uiService'].getUiConfiguration).toHaveBeenCalledTimes(0)
+      })
+    })
+    describe('When we require a configuration', () => {
+      it('the uiIdentifier is added to the pool ', () => {
+        service.uiConfReady('conf1').subscribe((conf) => (uiConf = conf))
+        expect(service['uiConfigurations']).toHaveProperty('conf1')
+      })
+      it('the api has been called ', () => {
+        expect(uiApiServiceMock.getUiConfiguration).toHaveBeenCalledWith(
+          'conf1'
+        )
+      })
+      it('return expected conf', () => {
+        expect(uiConf).toEqual(UI_FIXTURES)
+        jest.clearAllMocks()
+      })
+    })
+
+    describe('When we require configuration already fetched', () => {
+      beforeEach(() => {})
+      it('calls api only once', () => {
+        service.uiConfReady('conf1').subscribe((conf) => (uiConf = conf))
+        service.uiConfReady('conf1').subscribe((conf) => (uiConf = conf))
+        expect(service['uiService'].getUiConfiguration).toHaveBeenCalledTimes(1)
```

Comment:
@jahow each `subscribe` actually trigger a call to the `service['uiService'].getUiConfiguration` method, that should not be the case the the `shareReplay` should it ?
It probably does not trigger the Http call though...

Reply:
weird indeed, it should not work like that... needs some investigation

Reply:
Ok I understood ... 
The service is instanciated for each test, but the mock service is the same object, so a spy on the mock increases the call stack while `shareReplay` works as expected.
A `jest.clearAllMocks` might be usefull in `afterEach` for some cases.

Reply:
yes! I've noticed that in other tests. Using mocks initalized once for the whole test suite is misleading for this reason. Maybe we should add `clearAllMocks` at the end of every spec file...